src/ray/raylet/worker_pool.h

```diff
@@ -457,6 +461,9 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
     std::unordered_map<Process, TaskID> dedicated_workers_to_tasks;
     /// A map for speeding up looking up the pending worker for the given task.
     std::unordered_map<TaskID, Process> tasks_to_dedicated_workers;
+    /// A map for speeding up looking up the pending runtime env request for the given
+    /// task.
+    std::unordered_map<TaskID, std::string> tasks_to_pending_runtime_envs;
```

Comment:
The value is not used, we can just use a set instead of a map: ` std::unordered_set<TaskID> tasks_with_pending_runtime_envs;`
By the way, could you also change `tasks_to_dedicated_workers` to a set?